src/SourceBuild/content/eng/pipelines/templates/steps/publish-test-results-pr.yml

```diff
@@ -0,0 +1,207 @@
+        fork_url="https://github.com/${fork_repo}"
+        target_url="https://github.com/${target_repo}"
+
+        repo_dir="repo-${time}"
+        git clone "${fork_url}" "${repo_dir}" --depth 1
+        cd "${repo_dir}"
+        git remote add upstream "${target_url}"
+
+        target_branch=$(echo "${{ parameters.pullRequestTargetBranch }}" | sed 's/internal\///')
+        git fetch upstream "${target_branch}"
+
+        new_branch_name="${monthDayYear}-source-build-tests-${time}"
+        git checkout -b "${new_branch_name}" "upstream/${target_branch}"
+
+        git config --global user.name "dotnet-sb-bot"
+        git config --global user.email "dotnet-sb-bot@microsoft.com"
+      }
+
+      function UnionExclusions() {
+        original_test_file=$1
+        shift
+        updated_test_files="$@"
+
+        cat $updated_test_files | sort | uniq > temp.txt
+        awk 'NR==FNR { lines[$0] = 1; next } $0 in lines' temp.txt $original_test_file > temp2.txt && mv temp2.txt $original_test_file
+
+        rm temp.txt
+
+        git add $original_test_file
+      }
+
+      function IntersectionExclusions() {
+        original_test_file=$1
+        shift
+        updated_test_files="$@"
+
+        for file in $updated_test_files; do
+          grep -F -x -f $file $original_test_file > temp.txt && mv temp.txt $original_test_file
+        done
+
+        git add $original_test_file
+      }
+
+      function CopyNewFilesAndUpdateGit() {
+        destination_path=$1
+        shift
+        updated_files="$@"
+
+        relative_path=$(realpath --relative-to="$(pwd)" "$destination_path")
+
+        for updated_file in $updated_files; do
+          original_filename=$(basename $updated_file | sed 's/Updated//g')
+          touch $destination_path/$original_filename
+          cp $updated_file $destination_path/$original_filename
+          git add $destination_path/$original_filename
+
+          body+=$'\n'
+          body+=" - Could not find $original_filename in $relative_path."
+          body+=" The updated file has been added in this PR."
+          body+=$'\n'
+        done
+      }
+
+      function CreateLicenseBaseline() {
+        destination_path=$1
+        updated_test_file=$2
+        default_baseline_content=$'{\n  "files": []\n}'
+        baseline_path="$destination_path/baselines/licenses"
+
+        if [[ "$(cat $updated_test_file)" != $default_baseline_content ]]; then
+          CopyNewFilesAndUpdateGit $baseline_path $updated_test_file
+        fi
+      }
+
+      function MakePrChanges() {
+        relative_target_path="src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/assets"
+        absolute_target_path=$(realpath $relative_target_path)
+        declare -A files
+        updated_test_files=$(find "${{ parameters.testResultsPath }}" -name "Updated*");
+
+        for test_file in $updated_test_files; do
+          filename=$(basename "$test_file" | sed 's/Updated//g' | sed 's/\..*$//')
```

Comment:
Need a comment here explaining that this is stripping off everything after the first `.` in the filename. And it should be stated that this will cause exclusion files to be combined as their names will only differ in the content that was stripped away.